src/components/page-header/search-popover.tsx

```diff
@@ -0,0 +1,92 @@
+import { getCanvasSize, pointerPosToSVGCoord } from '../../util/helpers';
+import { useWindowSize } from '../../util/hooks';
+import { StationAttributes } from '../../constants/stations';
+
+interface SearchData {
+    id: StnId;
+    value: string;
+}
+
+export const SearchPopover = () => {
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    const dispatch = useRootDispatch();
+    const graph = React.useRef(window.graph);
+
+    const size = useWindowSize();
+    const { width, height } = getCanvasSize(size);
+
+    const [data, setData] = React.useState<SearchData[]>([]);
+    React.useEffect(() => {
+        if (isOpen) {
+            setData(
+                graph.current
+                    .nodes()
+                    .filter(node => node.startsWith('stn'))
+                    .map(stnId => {
+                        const attr = graph.current.getNodeAttributes(stnId);
+                        const type = attr.type;
+                        const name = (attr[type] as StationAttributes).names.join('/');
+                        return { id: stnId, value: name } as SearchData;
+                    })
+            );
+        }
+    }, [isOpen]);
+
+    const moveCanvas = (stnId: StnId) => {
+        dispatch(setSelected(new Set([stnId])));
+        const newSvgViewBoxZoom = Math.max(0, Math.min(400, -0.132 * height + 117.772));
+        const { x, y } = pointerPosToSVGCoord((width - 300) / 2, height / 2, newSvgViewBoxZoom, {
+            x: -graph.current.getNodeAttribute(stnId, 'x'),
+            y: -graph.current.getNodeAttribute(stnId, 'y'),
+        });
+        dispatch(setSvgViewBoxZoom(newSvgViewBoxZoom));
+        dispatch(setSvgViewBoxMin({ x: -x, y: -y }));
+    };
+
+    const fields: RmgFieldsField[] = [
+        {
+            type: 'custom',
+            label: '',
+            component: (
+                <RmgAutoComplete
+                    data={data}
+                    displayHandler={item => <Badge>{item.value}</Badge>}
+                    filter={(query, item) =>
+                        item.value.toLowerCase().includes(query.toLowerCase()) ||
+                        Object.values(item.value).some(name => name.toLowerCase().includes(query.toLowerCase()))
+                    }
+                    value=""
+                    onChange={item => moveCanvas(item.id)}
+                />
+            ),
+        },
+    ];
+
+    return (
+        <Popover isOpen={isOpen} onOpen={() => setIsOpen(true)} onClose={() => setIsOpen(false)}>
+            <PopoverTrigger>
+                <IconButton
+                    aria-label="zoom"
+                    variant="ghost"
+                    size="sm"
+                    icon={<MdSearch />}
+                    onClick={() => setIsOpen(!isOpen)}
+                />
+            </PopoverTrigger>
+            <PopoverContent>
+                <PopoverBody>
+                    <RmgFields fields={fields} noLabel />
```

Comment:
If we are using a custom field, then no `RmgFields` is required. We can use `RmgAutoComplete` here directly.